internal/cmd/skupper/site/site_create.go

```diff
@@ -0,0 +1,215 @@
+		if !ok {
+			validationErrors = append(validationErrors, fmt.Errorf("site name is not valid: %s", err))
+		}
+	}
+
+	if cmd.flags.linkAccessType != "" {
+		ok, err := linkAccessTypeValidator.Evaluate(cmd.flags.linkAccessType)
+		if !ok {
+			validationErrors = append(validationErrors, fmt.Errorf("link access type is not valid: %s", err))
+		}
+	}
+
+	if !cmd.flags.enableLinkAccess && len(cmd.flags.linkAccessType) > 0 {
+		validationErrors = append(validationErrors, fmt.Errorf("for the site to work with this type of linkAccess, the --enable-link-access option must be set to true"))
+	}
+
+	if cmd.flags.serviceAccount != "" {
+		ok, err := stringValidator.Evaluate(cmd.flags.serviceAccount)
+		if !ok {
+			validationErrors = append(validationErrors, fmt.Errorf("service account name is not valid: %s", err))
+		}
+		cmd.serviceAccountName = cmd.flags.serviceAccount
+	}
+
+	return validationErrors
+}
+
+func (cmd *CmdSiteCreate) InputToOptions() {
+
+	options := make(map[string]string)
+
+	options[site.SiteConfigNameKey] = cmd.siteName
+
+	if cmd.flags.enableLinkAccess {
+		if cmd.flags.linkAccessType == "" {
+			options[site.SiteConfigIngressKey] = "loadbalancer"
+		} else {
+			options[site.SiteConfigIngressKey] = cmd.flags.linkAccessType
+		}
+	} else {
+		options[site.SiteConfigIngressKey] = "none"
+	}
+
+	if cmd.flags.linkAccessHost != "" {
+		options[site.SiteConfigIngressHostKey] = cmd.flags.linkAccessHost
+	}
+
+	cmd.options = options
+
+}
+
+func (cmd *CmdSiteCreate) Run() error {
+
+	resource := v1alpha1.Site{
+		ObjectMeta: metav1.ObjectMeta{Name: cmd.siteName},
+		Spec: v1alpha1.SiteSpec{
+			Settings:       cmd.options,
+			ServiceAccount: cmd.serviceAccountName,
+		},
+	}
+
+	_, err := cmd.Client.Sites(cmd.Namespace).Create(context.TODO(), &resource, metav1.CreateOptions{})
+	return err
+}
+
+func (cmd *CmdSiteCreate) WaitUntilReady() error {
+
+	err := utils.NewSpinner("Waiting for site...", 5, func() error {
+
+		resource, err := cmd.Client.Sites(cmd.Namespace).Get(context.TODO(), cmd.siteName, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+
+		if resource != nil && resource.Status.StatusMessage == "OK" {
+			return nil
+		}
+
+		return fmt.Errorf("error getting the resource")
+	})
+
+	if err != nil {
+		return fmt.Errorf("Site \"%s\" not ready yet, check the logs for more information\n", cmd.siteName)
+	}
+
+	err = utils.NewSpinner("Waiting for status...", 5, func() error {
+
+		configmap, err := cmd.KubeClient.CoreV1().ConfigMaps(cmd.Namespace).Get(context.TODO(), types.NetworkStatusConfigMapName, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+
+		if configmap != nil && configmap.Data != nil && len(configmap.Data) > 0 {
+			return nil
+		}
+
+		return fmt.Errorf("error getting the status updated")
+	})
+
+	if err != nil {
+		fmt.Println("Status is not ready yet, check the logs for more information")
+	}
+
+	fmt.Printf("Site \"%s\" is ready\n", cmd.siteName)
```

Comment:
Suggest updating 'as not ready' will be followed by 'is ready' output

Reply:
Fixed in 5876a50a111ee1f6ce9ce393b5b82bee5f71b8a7